### DISCUSSION TITLE
FreeRADIUS compatibility

### DISCUSSION THREAD

**Discussion by baba2k** [lldap/lldap#559] [Q&A]
Hi,

ist LLDAP compatible with FreeRADIUS PEAP + EAP-MSCHAPv2 ? 
http://deployingradius.com/documents/protocols/compatibility.html

Passwords must be stored in CLEARTEXT (or NT hash) on LDAP server.

Thank you!

Best, baba

**Comment by nitnelave** [comment]
Hi! Unfortunately, no. Rather than receiving the password and then hashing it, LLDAP uses the OPAQUE protocol to identify users: when a user changes their password, the server only receives enough information to do a zero-knowledge proof later on that the user has the same password. In particular, the server never receives the password or its hash.

That makes it incompatible with any other service relying on password hashes for authentication, LLDAP is the only service able to authenticate the user.

Maybe you could benefit from keycloak, authelia or go-auth (I don't remember exactly that service's name)? They can provide modern authentication protocol while relying on LLDAP for the actual authentication.
